File: ast/enum_definition.go
package ast

type EnumDefinition struct {
	Name *Identifier
	Values []*Identifier
}

func (n *EnumDefinition) Accept(visitor Visitor) {
	visitor.VisitEnumDefinition(n)
}

func (n *EnumDefinition) DefinedName() string {
	return n.Name.Name
}

File: src/main/java/com/moybl/ssdl/ast/EnumDefinition.java
package com.moybl.ssdl.ast;

import com.moybl.ssdl.Position;

import java.util.List;

public class EnumDefinition extends Definition {

	private Identifier name;
	private List<Identifier> values;

	public EnumDefinition(Position position, Identifier name, List<Identifier> values) {
		super(position);
		this.name = name;
		this.values = values;
	}

	public void accept(Visitor visitor) {
		visitor.visit(this);
	}

	public Identifier getName() {
		return name;
	}

	public String getDefinedName() {
		return name.getName();
	}

	public List<Identifier> getValues() {
		return values;
	}

}
